homeassistant/components/twinkly/select.py

```diff
@@ -0,0 +1,63 @@
+"""The Twinkly select component."""
+
+from __future__ import annotations
+
+import logging
+
+from ttls.client import TWINKLY_MODES
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TwinklyConfigEntry, TwinklyCoordinator
+from .const import DEV_MODEL, DEV_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TwinklyConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a mode select from a config entry."""
+    entity = TwinklyModeSelect(config_entry.runtime_data)
+    async_add_entities([entity], update_before_add=True)
+
+
+class TwinklyModeSelect(CoordinatorEntity[TwinklyCoordinator], SelectEntity):
+    """Twinkly Mode Selection."""
+
+    _attr_has_entity_name = True
+    _attr_name = "Mode"
+    _attr_options = TWINKLY_MODES
+
+    def __init__(self, coordinator: TwinklyCoordinator) -> None:
+        """Initialize TwinklyModeSelect."""
+        super().__init__(coordinator)
+        device_info = coordinator.data.device_info
+        mac = device_info["mac"]
+
+        self._attr_unique_id = f"{mac}_mode"
+        self._attr_device_info = DeviceInfo(
```

Comment:
I thought about that too, but didn't want to change too much in a single pull request. Instead, I would make a subsequent pull request. Unless that is a mandatory requirement for the code to be merged.

Reply:
I think that ideally it would have been in a PR that enters before this one does, so that this already uses the new entity class instead of duplicating the DeviceInfo def and later removing it. Are you ok with opening that PR before this one gets in?